pvactools/lib/generate_reviews_files.py

```diff
@@ -50,7 +52,62 @@ def make_column_unique(df, column_name):
     return df
 
 
-def main(peptides_path, classI_path, classII_path, sample_name, all_epitopes_flag):
+# Function that reads a vcf and returns the variants that contain "PASS" in "FILTER"
+def load_pass_variants(vcf_path):
+    variants = set()
+    reader = vcfpy.Reader.from_path(vcf_path)
+    for record in reader:
+        if record.FILTER and record.FILTER[0] != "PASS":
+            continue
+        chrom = record.CHROM
+        ref = record.REF
+        for alt in record.ALT:
+            alt_str = alt.value
+            start = record.affected_start
+            end = record.affected_end
+            variants.add((chrom, start, end, ref, alt_str))
+    return variants
+
+
+# Function that fills the "Variant Called in External VCF" column based on the presence
+# of the classI_tsv variant in the provided VCF along with a value of "PASS" in "FILTER"
+def fill_variant_called_column(df, input_vcf, external_vcf):
+    input_variants = load_pass_variants(input_vcf)
+    external_variants = load_pass_variants(external_vcf)
+
+    called = []
+    for _, row in df.iterrows():
+        parts = row["ID"].split("-")
+        chrom = parts[0]
+        start = int(parts[1])
+        end = int(parts[2])
+        ref = parts[3]
+        alt = parts[4]
+        variant_key = (chrom, start, end, ref, alt)
+
+        if variant_key in input_variants and variant_key in external_variants:
+            called.append(True)
+        else:
+            called.append(False)
+    df["Variant Called in External VCF"] = called
+
+
+def main(peptides_path, classI_path, classII_path, input_vcf, external_vcf, sample_name,
+        all_epitopes_flag, allowed_evaluations, output_file_prefix, output_path):
+    reviewed_candidates = pd.read_csv(classI_path, sep="\t")
+    reviewed_candidates = reviewed_candidates[reviewed_candidates["Evaluation"].isin(allowed_evaluations)]
```

Comment:
From what Malachi was saying, I'm not sure whether this is supposed to subset the reviewed_candidates to only the ones with the right Evaluation. Might want to get clarification on that.